locksmith/src/operations/userOperations.ts

```diff
@@ -0,0 +1,41 @@
+const ethJsUtil = require('ethereumjs-util')
+const models = require('../models')
+const { User, UserReference } = models
+import * as types from '../types'
+
+class UserOperations {
+  static async createUser(input: types.UserCreationInput): Promise<Boolean> {
+    let user: any = await User.create({
+      publicKey: ethJsUtil.toChecksumAddress(input.publicKey),
+      recoverySeed: input.recoverySeed,
+      encryptedPrivateKey: input.encryptedPrivateKey,
+    })
+
+    let userReference = await UserReference.create({
+      emailAddress: input.emailAddress.toLowerCase(),
+    })
+
+    if (userReference) {
+      userReference.setUser(user)
```

Comment:
Could we do that above directly when creating the object? This would save at least on SQL request?

Reply:
Agreed, i'll put this and the other call in a transaction. I've started down this direction but paused due to some of the ripples caused